### PR TITLE
[receiver/hostmetrics] Hostmetrics receiver scrape all processes on linux

### PR DESCRIPTION
**Pull request by jskiba**
**Description:** <Describe what has changed.>
This feature would allow scraping all processes on Linux machines without root privileges.
Feature adds two extra options in `receiver/hostmetrics`:
- `mute_process_exe_error`
- `mute_process_io_error`

**Link to tracking Issue:** [18923](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18923)

**Testing:** Testing was performed manually by comparing lists of scraped processes on Ubuntu 22 VM. The list of processes scraped by `otelcontribcol` built from this branch included processes that were not gathered and logged out by `otelcontribcol` compiled from the main branch.

**Documentation:** Added mentioned flags to README of `receiver/hostmetrics`.

### CODE REVIEW

**Comment by jskiba** [comment]
Ok, should be working now, at least tests pass here https://github.com/jskiba/opentelemetry-collector-contrib/pull/1

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go [hunk 1]
```diff
@@ -288,7 +297,7 @@ func (s *scraper) scrapeAndAppendMemoryUtilizationMetric(now pcommon.Timestamp,
 }
 
 func (s *scraper) scrapeAndAppendDiskMetrics(now pcommon.Timestamp, handle processHandle) error {
-	if !(s.config.MetricsBuilderConfig.Metrics.ProcessDiskIo.Enabled || s.config.MetricsBuilderConfig.Metrics.ProcessDiskOperations.Enabled) || runtime.GOOS == "darwin" {
+	if !(s.config.MetricsBuilderConfig.Metrics.ProcessDiskIo.Enabled || s.config.MetricsBuilderConfig.Metrics.ProcessDiskOperations.Enabled) || s.config.MuteProcessIOError || runtime.GOOS == "darwin" {
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go:300]: Is this an intended change? So we don't even attempt to scrape disk process metrics if `mute_process_io_error=true`?

If so, why do we need another configuration option just to disable collection disk process metrics?
Users should be able to achieve the same with:
```
metrics:
  process.disk.io:
    enabled: false
  process.disk.operations:
    enabled: false
```

Comment (jskiba) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go]: Oh, I meant to mute these errors, correction in the next commit
```
io, err := handle.IOCounters()
if err != nil {
	if s.config.MuteProcessIOError {
		return nil
        }
	return err
}
```

**Comment by jskiba** [comment]
Some `sshcheckreceiver` tests failed

**Review by dmitryax — APPROVED** [review]

**Pull request merged**